types/testing/id_test.cc

```diff
@@ -51,11 +51,29 @@ BOOST_AUTO_TEST_CASE( test_basic_id )
 
 BOOST_AUTO_TEST_CASE( test_uuid_id )
 {
+    // lower case
     string uuid = "0828398c-5965-11e0-84c8-0026b937c8e1";
     Id id(uuid);
     BOOST_CHECK_EQUAL(id.type, Id::UUID);
     BOOST_CHECK_EQUAL(id.toString(), uuid);
     checkSerializeReconstitute(id);
+
```

Comment:
I would add one test case that the hash from an old capitalized UUID and a new capitialized UUID are the same.  You will need to record one and its hash from a branch without the change to get the test case working.